### ISSUE TITLE
issue when entering xonsh multiple times

### ISSUE THREAD

**Issue by mitnk** [xonsh/xonsh#2198]
current master; repeat enter `xonsh` in `xonsh` will got errors:
```
$ xonsh
$ xonsh
$ xonsh
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/xonsh/base_shell.py", line 322, in default
    run_compiled_code(code, self.ctx, None, 'single')
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/xonsh/codecache.py", line 65, in run_compiled_code
    func(code, glb, loc)
  File "<xonsh-code>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/xonsh/built_ins.py", line 856, in subproc_captured_hiddenobject
    return run_subproc(cmds, captured='hiddenobject')
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/xonsh/built_ins.py", line 797, in run_subproc
    command = HiddenCommandPipeline(specs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/xonsh/proc.py", line 1704, in __init__
    if update_fg_process_group(pipeline_group, background):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/xonsh/proc.py", line 1640, in update_fg_process_group
    return give_terminal_to(pipeline_group)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/xonsh/jobs.py", line 139, in give_terminal_to
    os.tcsetpgrp(sys.stderr.fileno(), pgid)
OSError: [Errno 25] Inappropriate ioctl for device
```

This issue is introduced both by #2177 and #2191 - revert changes of #2191 will fix the issue (I'm not saying this is the solution thou), but I need to check what really happening and how to fix it.

**Comment by scopatz** [comment]
`sys.stderr.fileno()` should probably be replaced with `2` for stability reasons. Thanks for reporting this, @mitnk.

**Comment by mitnk** [comment]
In command_cache we have
```
predictors = {
    ...
    'xonsh': predict_shell,
    ...
}
```

**Comment by mitnk** [comment]
So revert #2191 seems a good solution. we can try other ways to fix #2186 (if it still exist with #2177 merged)

**Comment by mitnk** [comment]
Fixed by #2211.
